lib/vsc/utils/fancylogger.py

```diff
@@ -188,12 +192,20 @@ def makeRecord(self, name, level, pathname, lineno, msg, args, excinfo, func=Non
             new_msg = msg.encode('utf8', 'replace')
         return logrecordcls(name, level, pathname, lineno, new_msg, args, excinfo)
 
+    def fail(self, message, *args):
+        """Log error message and raise exception."""
+        formatted_message = message % args
+        self.DEFAULT_RAISE_EXCEPTION_LOG_METHOD(formatted_message)
```

Comment:
indeed, and why let the logger reformat the string, we already have a formatted version